chunks/chunks.go

```diff
@@ -57,8 +57,10 @@ type Meta struct {
 }
 
 // writeHash writes the chunk encoding and raw data into the provided hash.
-func (cm *Meta) writeHash(h hash.Hash) error {
-	if _, err := h.Write([]byte{byte(cm.Chunk.Encoding())}); err != nil {
+// 'buf' byte slice is used to avoid allocating new byte slice. Assumes that len(buf) is 0.
+func (cm *Meta) writeHash(h hash.Hash, buf []byte) error {
+	buf = append(buf, byte(cm.Chunk.Encoding()))
```

Comment:
Instead of assuming the buffer always has zero length, you could explicitly use `buf[:0]`.

Reply:
As it was internal function, I just assumed it will be passed with 0 len. I will change it to `buf[:0]`.